### PR TITLE
Builder - Content Editor - Web page (C039 - #32)

### PR DESCRIPTION
**Pull request by kboczkowski-clurgo**
## Description
In this PR I've addeed create new geostory component mechanism - WebPage

## Issues
 - #4514

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)

 - [x] Feature

**What is the current behavior?** (You can also link to an open issue here)

**What is the new behavior?**

**Does this PR introduce a breaking change?** (check one with "x", remove the other)

 - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

**Comment by tdipisa** [comment]
@kboczkowski-clurgo please fix the issues identified by @MV88, the Web Page content must be added not only as a additional content of an existing section (immersive or not) but also as a section per se like the other ones. The requirement in the [issue](https://github.com/geosolutions-it/MapStore2-C039/issues/32) description is clear about that.

**Comment by tdipisa** [comment]
>@tdipisa I noticed that you cannot add directly a web page content, you have to add a media section or paragraph

@kboczkowski-clurgo looking at the last commit message I suppose that latest updates don't fix this problem but only the consol error. Isn't it?

**Comment by kboczkowski-clurgo** [comment]
@tdipisa actualy I've fixed problem with editing which @MV88 discover. Now I'm working on adding this component via section directly.

**Comment by kboczkowski-clurgo** [comment]
https://github.com/geosolutions-it/MapStore2/pull/4600#pullrequestreview-328832449
It's done

**Comment by tdipisa** [comment]
> Hi, i was not able to load these urls:
> 
> * http://localhost:8081/?debug=true#/geostory/newgeostory/
> * http://localhost:8081/#/geostory/newgeostory/

This doesn't make sens to me, why we should support these kind of URLs? 

Maybe what we can do here is just to make the regex configurable from plugin config 

@offtherailz what do you think?

**Comment by offtherailz** [comment]
the only problem concerns "localhost", because the regex denies domain names without a point. This may fail with host files or local DNS. 

Because the system may need to satisfy different requirements, as @tdipisa suggested, I should allow to configure the regex (2nd param of isValidURL may be the regex, by default the one currently used, maybe without the `.` check, but this is not necessary, if configurable. 
From outside, (where isValidURL is called) you can get it from ConfigUtils.getConfigProp("GeoStoryValidIframeURLRegex")  and pass it as 2nd param. I think it's enough.

**Comment by kboczkowski-clurgo** [comment]
@MV88 @tdipisa @offtherailz PR is updated with changes suggested by @offtherailz. Now validation util has optional parameter to pass custom regexp.

File: web/client/components/geostory/contents/WebPageWrapper.jsx [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+        );
+    }
+
+    updateURL = ({ target: { value: url }}) => {
+        this.setState({ url });
+    }
+
+    save = () => {
+        const { url } = this.state;
+        const error = !isValidURL(url);
```
Comment (MV88) [inline web/client/components/geostory/contents/WebPageWrapper.jsx]: can you pass ConfigUtils.getConfigProp("GeoStoryValidIframeURLRegex") as second argument? 
in that case we can override it anytime without changing the code

Verify that if the property is missing in config, then it uses the default one

File: web/client/components/geostory/contents/WebPageWrapper.jsx [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+        this.setState({ error: !isValidURL(url) });
```
Comment (MV88) [inline web/client/components/geostory/contents/WebPageWrapper.jsx]:
```suggestion
        this.setState({ error });
```

File: web/client/components/geostory/contents/WebPageWrapper.jsx (hunk 1 above)
Comment (kboczkowski-clurgo) [reply web/client/components/geostory/contents/WebPageWrapper.jsx]: ok

Comment (kboczkowski-clurgo) [reply web/client/components/geostory/contents/WebPageWrapper.jsx]: @MV88 done

**Pull request merged**